From 32d8dd492790c5eec9f0e8ffbf681d4ee057312e Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Tue, 13 Sep 2011 16:06:04 +0200
Subject: [PATCH 064/154] pvrsink: move to pvr plugin

Move pvrsink to the pvr plugin. Rename gstducati.* to gstpvr.*
Make pvrsink use its own debug category.
---
 sys/pvr2d/Makefile.am        |    2 +-
 sys/pvr2d/gstducati.c        |   87 ------------------------------------------
 sys/pvr2d/gstducati.h        |   45 ---------------------
 sys/pvr2d/gstpvr.c           |   85 +++++++++++++++++++++++++++++++++++++++++
 sys/pvr2d/gstpvr.h           |   42 ++++++++++++++++++++
 sys/pvr2d/gstpvrbufferpool.h |    2 +-
 sys/pvr2d/gstpvrvideosink.c  |    1 +
 7 files changed, 130 insertions(+), 134 deletions(-)
 delete mode 100644 sys/pvr2d/gstducati.c
 delete mode 100644 sys/pvr2d/gstducati.h
 create mode 100644 sys/pvr2d/gstpvr.c
 create mode 100644 sys/pvr2d/gstpvr.h

diff --git a/sys/pvr2d/Makefile.am b/sys/pvr2d/Makefile.am
index c9cd64f..9823282 100644
--- a/sys/pvr2d/Makefile.am
+++ b/sys/pvr2d/Makefile.am
@@ -1,7 +1,7 @@
 plugin_LTLIBRARIES = libgstpvr.la
 
 libgstpvr_la_SOURCES = \
-	gstducati.c \
+	gstpvr.c \
 	gstpvrbufferpool.c \
 	gstpvrvideosink.c
 
diff --git a/sys/pvr2d/gstducati.c b/sys/pvr2d/gstducati.c
deleted file mode 100644
index 50e2954..0000000
--- a/sys/pvr2d/gstducati.c
+++ /dev/null
@@ -1,87 +0,0 @@
-/*
- * GStreamer
- * Copyright (c) 2010, Texas Instruments Incorporated
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation
- * version 2.1 of the License.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
- */
-
-#ifdef HAVE_CONFIG_H
-#  include <config.h>
-#endif
-
-#include "gstducati.h"
-#include "gstpvrvideosink.h"
-
-GST_DEBUG_CATEGORY (gst_ducati_debug);
-
-static gboolean
-plugin_init (GstPlugin * plugin)
-{
-  GST_DEBUG_CATEGORY_INIT (gst_ducati_debug, "ducati", 0, "ducati");
-
-  /* TODO .. find some way to reasonably detect if the corresponding
-   * codecs are actually available..
-   */
-  return gst_element_register (plugin, "pvrvideosink", GST_RANK_PRIMARY,
-      GST_TYPE_PVRVIDEOSINK);
-}
-
-void *
-gst_ducati_alloc_1d (gint sz)
-{
-  MemAllocBlock block = {
-    .pixelFormat = PIXEL_FMT_PAGE,
-    .dim.len = sz,
-  };
-  return MemMgr_Alloc (&block, 1);
-}
-
-void *
-gst_ducati_alloc_2d (gint width, gint height, guint * sz)
-{
-  MemAllocBlock block[] = { {
-          .pixelFormat = PIXEL_FMT_8BIT,
-          .dim = {.area = {
-                      .width = width,
-                      .height = ALIGN2 (height, 1),
-                  }},
-      .stride = 4096}, {
-        .pixelFormat = PIXEL_FMT_16BIT,
-        .dim = {.area = {
-                    .width = width,
-                    .height = ALIGN2 (height, 1) / 2,
-                }},
-      .stride = 4096}
-  };
-  if (sz) {
-    *sz = (4096 * ALIGN2 (height, 1) * 3) / 2;
-  }
-  return MemMgr_Alloc (block, 2);
-}
-
-/* PACKAGE: this is usually set by autotools depending on some _INIT macro
- * in configure.ac and then written into and defined in config.h, but we can
- * just set it ourselves here in case someone doesn't use autotools to
- * compile this code. GST_PLUGIN_DEFINE needs PACKAGE to be defined.
- */
-#ifndef PACKAGE
-#  define PACKAGE "ducati"
-#endif
-
-GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
-    GST_VERSION_MINOR,
-    "ducati",
-    "Hardware accellerated codecs for OMAP4",
-    plugin_init, VERSION, "LGPL", "GStreamer", "http://gstreamer.net/")
diff --git a/sys/pvr2d/gstducati.h b/sys/pvr2d/gstducati.h
deleted file mode 100644
index a90af35..0000000
--- a/sys/pvr2d/gstducati.h
+++ /dev/null
@@ -1,45 +0,0 @@
-/*
- * GStreamer
- * Copyright (c) 2010, Texas Instruments Incorporated
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation
- * version 2.1 of the License.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
- */
-
-#ifndef __GST_DUCATI_H__
-#define __GST_DUCATI_H__
-
-#include <stdint.h>
-#include <string.h>
-
-#include <tiler.h>
-#include <tilermem.h>
-#include <memmgr.h>
-
-#include <gst/gst.h>
-
-G_BEGIN_DECLS
-
-GST_DEBUG_CATEGORY_EXTERN (gst_ducati_debug);
-#define GST_CAT_DEFAULT gst_ducati_debug
-
-/* align x to next highest multiple of 2^n */
-#define ALIGN2(x,n)   (((x) + ((1 << (n)) - 1)) & ~((1 << (n)) - 1))
-
-void * gst_ducati_alloc_1d (gint sz);
-void * gst_ducati_alloc_2d (gint width, gint height, guint * sz);
-
-G_END_DECLS
-
-#endif /* __GST_DUCATI_H__ */
diff --git a/sys/pvr2d/gstpvr.c b/sys/pvr2d/gstpvr.c
new file mode 100644
index 0000000..e440e95
--- /dev/null
+++ b/sys/pvr2d/gstpvr.c
@@ -0,0 +1,85 @@
+/*
+ * GStreamer
+ * Copyright (c) 2010, Texas Instruments Incorporated
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation
+ * version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#include "gstpvr.h"
+#include "gstpvrvideosink.h"
+
+GST_DEBUG_CATEGORY (gst_debug_pvrvideosink);
+
+static gboolean
+plugin_init (GstPlugin * plugin)
+{
+  GST_DEBUG_CATEGORY_INIT (gst_debug_pvrvideosink, "pvrvideosink", 0,
+      "pvrvideosink");
+
+  return gst_element_register (plugin, "pvrvideosink", GST_RANK_PRIMARY,
+      GST_TYPE_PVRVIDEOSINK);
+}
+
+void *
+gst_ducati_alloc_1d (gint sz)
+{
+  MemAllocBlock block = {
+    .pixelFormat = PIXEL_FMT_PAGE,
+    .dim.len = sz,
+  };
+  return MemMgr_Alloc (&block, 1);
+}
+
+void *
+gst_ducati_alloc_2d (gint width, gint height, guint * sz)
+{
+  MemAllocBlock block[] = { {
+          .pixelFormat = PIXEL_FMT_8BIT,
+          .dim = {.area = {
+                      .width = width,
+                      .height = ALIGN2 (height, 1),
+                  }},
+      .stride = 4096}, {
+        .pixelFormat = PIXEL_FMT_16BIT,
+        .dim = {.area = {
+                    .width = width,
+                    .height = ALIGN2 (height, 1) / 2,
+                }},
+      .stride = 4096}
+  };
+  if (sz) {
+    *sz = (4096 * ALIGN2 (height, 1) * 3) / 2;
+  }
+  return MemMgr_Alloc (block, 2);
+}
+
+/* PACKAGE: this is usually set by autotools depending on some _INIT macro
+ * in configure.ac and then written into and defined in config.h, but we can
+ * just set it ourselves here in case someone doesn't use autotools to
+ * compile this code. GST_PLUGIN_DEFINE needs PACKAGE to be defined.
+ */
+#ifndef PACKAGE
+#  define PACKAGE "ducati"
+#endif
+
+GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
+    GST_VERSION_MINOR,
+    "pvr",
+    "Hardware accellerated codecs for OMAP4",
+    plugin_init, VERSION, "LGPL", "GStreamer", "http://gstreamer.net/")
diff --git a/sys/pvr2d/gstpvr.h b/sys/pvr2d/gstpvr.h
new file mode 100644
index 0000000..d2c57af
--- /dev/null
+++ b/sys/pvr2d/gstpvr.h
@@ -0,0 +1,42 @@
+/*
+ * GStreamer
+ * Copyright (c) 2010, Texas Instruments Incorporated
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation
+ * version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef __GST_DUCATI_H__
+#define __GST_DUCATI_H__
+
+#include <stdint.h>
+#include <string.h>
+
+#include <tiler.h>
+#include <tilermem.h>
+#include <memmgr.h>
+
+#include <gst/gst.h>
+
+G_BEGIN_DECLS
+
+/* align x to next highest multiple of 2^n */
+#define ALIGN2(x,n)   (((x) + ((1 << (n)) - 1)) & ~((1 << (n)) - 1))
+
+void * gst_ducati_alloc_1d (gint sz);
+void * gst_ducati_alloc_2d (gint width, gint height, guint * sz);
+
+G_END_DECLS
+
+#endif /* __GST_DUCATI_H__ */
diff --git a/sys/pvr2d/gstpvrbufferpool.h b/sys/pvr2d/gstpvrbufferpool.h
index 678ae73..b73fddc 100644
--- a/sys/pvr2d/gstpvrbufferpool.h
+++ b/sys/pvr2d/gstpvrbufferpool.h
@@ -21,7 +21,7 @@
 #ifndef __GSTPVRBUFFERPOOL_H__
 #define __GSTPVRBUFFERPOOL_H__
 
-#include "gstducati.h"
+#include "gstpvr.h"
 
 G_BEGIN_DECLS
 
diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index b4fb8b6..e3ec4dd 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -34,6 +34,7 @@
 #define DEFAULT_MIN_QUEUED_BUFS 1
 
 GST_DEBUG_CATEGORY_EXTERN (gst_debug_pvrvideosink);
+#define GST_CAT_DEFAULT gst_debug_pvrvideosink
 
 static void gst_pvrvideosink_reset (GstPVRVideoSink * pvrvideosink);
 static GstFlowReturn gst_pvrvideosink_buffer_alloc (GstBaseSink * bsink,
-- 
1.7.5.4

